sky/backends/cloud_vm_ray_backend.py

```diff
@@ -737,6 +738,32 @@ def _update_blocklist_on_azure_error(self, region, zones, stdout, stderr):
         else:
             self._blocked_regions.add(region.name)
 
+    def _update_blocklist_on_lambda_error(self, region, zones, stdout, stderr):
+        del zones  # Unused.
+        style = colorama.Style
+        stdout_splits = stdout.split('\n')
+        stderr_splits = stderr.split('\n')
+        errors = [
+            s.strip()
+            for s in stdout_splits + stderr_splits
+            if 'LambdaLabsError:' in s.strip()
+        ]
+        if not errors:
+            logger.info('====== stdout ======')
+            for s in stdout_splits:
+                print(s)
+            logger.info('====== stderr ======')
+            for s in stderr_splits:
+                print(s)
+            with ux_utils.print_exception_no_traceback():
+                raise RuntimeError('Errors occurred during provision; '
+                                   'check logs above.')
+
+        logger.warning(f'Got error(s) in {region.name}:')
+        messages = '\n\t'.join(errors)
+        logger.warning(f'{style.DIM}\t{messages}{style.RESET_ALL}')
+        self._blocked_regions.add(region.name)
```

Comment:
nit: does lambdas lab provides some specific error message when the resources is not available in all the regions so that we can block all the regions after trying one region to speed up the failover (similar to what we did for GCP [here](https://github.com/ewzeng/skypilot/blob/61f3ccd4f70e6be288995816094965f71a7d6bb0/sky/backends/cloud_vm_ray_backend.py#L602-L611))? Otherwise, `sky launch --gpus V100:8` will take a long time before failing over to another cloud.

Reply:
Not in the error message, but you can find out if a resource is not available in multiple regions by running `ls`. Let me try to implement this optimization.

Reply:
Is there any easy way to test this GCP failover code? I suspect the fine-grain optimizer introduced a bug here.

Reply:
> Not in the error message, but you can find out if a resource is not available in multiple regions by running LambdaLabsClient.ls(). Let me try to implement this optimization.

It is fine to keep the current way, if the optimization is not easy. Just a good to have.

> Is there any easy way to test this GCP failover code? I suspect the fine-grain optimizer introduced a bug here.

Some high-end CPU/GPU machine may work here. Could you be more specific about the bug, cc @WoosukKwon ?

Reply:
I implemented the optimization, but it doesn't work because of the following bug (I think):

1. The fine grain optimizer sets `to_provision.region`.
2. Because `to_provision.region` is set, the loop [here](https://github.com/skypilot-org/skypilot/blob/786fcb668a0610c52d4091e91ebf089a962769c9/sky/backends/cloud_vm_ray_backend.py#L1126) only tries one region before exiting the loop. Updating the blocklist is useless because we clear the blocklist the next time we run this loop.

I don't think this bug is specific to Lambda Labs; it should be more general.

Reply:
Not sure if you checked the code below, we will reset the `to_provision` variable by optimizing again, i.e. the region/zone will be updated, and the `blocked_list` should not be cleared for different regions?

Reply:
Hmm. The region blocklist is cleared everytime `_retry_region_zones` is [called](https://github.com/skypilot-org/skypilot/blob/ad6296b71f8c6dde35ed47083e920606fb4e1b00/sky/backends/cloud_vm_ray_backend.py#L1087), so it's in `_retry_region_zones` where updating the region blocklist even matters. Your code snippet updates `_blocked_launchable_resources`, not the region blocklist.

Reply:
Ahh, great catch! That is a bug. Thanks for pointing that out!